core/src/main/scala/org/apache/spark/status/api/v1/AllStagesResource.scala

```diff
@@ -59,6 +59,9 @@ private[v1] object AllStagesResource {
       stageUiData: StageUIData,
       includeDetails: Boolean): StageData = {
 
+    val firstTaskLaunchedTime =
+      new Date(stageUiData.taskData.values.map(_.taskInfo.launchTime).filter(_ > 0).min)
```

Comment:
I think you need to guard against calling min on an empty list, eg. if no tasks have been launched yet.  So the type should probably be an Option